src/app/modules/item/helpers/item-permissions-constraints.ts

```diff
@@ -0,0 +1,221 @@
+import { ValidationErrors, ValidatorFn, AbstractControl } from '@angular/forms';
+import { GroupPermissions } from 'src/app/shared/http-services/group-permissions.service';
+import { PermissionsInfo, canGrantViewValues, canViewValues, permissionsInfoString } from './item-permissions';
+
+const bolden = (text: string): string => `<b>${text}</b>`;
```

Comment:
IMO, this file is violating a bit software engineering rules by mixing roles:
1. it finds constraint violations in permissions
2. it convert these violations in texts
3. in format it (this bold)

This should be done at 3 different places. Actually 1+2 is still ok, you can maybe leave it as it is, but not 3.

Reply:
fixed, the `validate...()` functions now return an object describing the constraint that is not met. This object is latter transformed into a readable error message